File: ORIGIN_Challenge_Frontend/ORIGIN_Challenge_Angular16/src/app/services/operaciones.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { tap, Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class OperacionesService {
  constructor(private http: HttpClient) { }

  baseUrl = "https://localhost:7112";

  Balance(numeroTarjeta: string): Observable<any> {
    return this.http.get<any>(`${this.baseUrl}/Operaciones/Balance?numeroTarjeta=${numeroTarjeta}`).pipe(tap((res: any) => { }));
  }

  Retiro(numeroTarjeta: string, cantidadRetiro: string): Observable<any> {
    return this.http.get<any>(`${this.baseUrl}/Operaciones/Retiro?numeroTarjeta=${numeroTarjeta}&cantidadRetiro=${cantidadRetiro}`).pipe(tap((res: any) => { }));
  }
}

File: ORIGIN_Challenge_Frontend/ORIGIN_Challenge_Blazor/Services/OperacionesService.cs
namespace ORIGIN_Challenge_Blazor.Services
{
    public class OperacionesService
    {
        private readonly HttpClient _httpClient;

        public OperacionesService(HttpClient httpClient)
        {
            _httpClient = httpClient;
        }

        public async Task<HttpResponseMessage> Balance(string numeroTarjeta)
        {
            return await _httpClient.GetAsync($"/Operaciones/Balance?numeroTarjeta={numeroTarjeta}");
        }

        public async Task<HttpResponseMessage> Retiro(string numeroTarjeta, string cantidadRetiro)
        {
            return await _httpClient.GetAsync($"/Operaciones/Retiro?numeroTarjeta={numeroTarjeta}&cantidadRetiro={cantidadRetiro}");
        }
    }
}
